sos/report/plugins/charmed_postgresql.py

```diff
@@ -0,0 +1,216 @@
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from typing import Optional
+import yaml
+
+from sos.report.plugins import Plugin, UbuntuPlugin
+
+SNAP_COMMON_PATH = "/var/snap/charmed-postgresql/common"
+SNAP_CURRENT_PATH = "/var/snap/charmed-postgresql/current"
+
+PATHS = {
+    "POSTGRESQL_CONF": SNAP_COMMON_PATH + "/var/lib/postgresql",
+    "POSTGRESQL_LOGS": SNAP_COMMON_PATH + "/var/log/postgresql",
+    "PATRONI_CONF": SNAP_CURRENT_PATH + "/etc/patroni",
+    "PATRONI_LOGS": SNAP_COMMON_PATH + "/var/log/patroni",
+    "PGBACKREST_CONF": SNAP_CURRENT_PATH + "/etc/pgbackrest",
+    "PGBACKREST_LOGS": SNAP_COMMON_PATH + "/var/log/pgbackrest",
+    "PGBOUNCER_CONF": SNAP_CURRENT_PATH + "/etc/pgbouncer",
+    "PGBOUNCER_LOGS": SNAP_COMMON_PATH + "/var/log/pgbouncer",
+}
+
+RUN_AS_SNAP_DAEMON = "runuser -u snap_daemon --"
+PSQL = "charmed-postgresql.psql"
+PATRONICTL = "charmed-postgresql.patronictl"
+
+
+class CharmedPostgreSQL(Plugin, UbuntuPlugin):
+
+    short_desc = "Charmed PostgreSQL"
+    plugin_name = "charmed_postgresql"
+    packages = ('charmed-postgresql',)
+
+    @property
+    def patroni_cluster_name(self) -> Optional[str]:
+        with open(f"{PATHS['PATRONI_CONF']}/patroni.yaml") as file:
```

Comment:
We need to enclose this in a try/Except to catch if the file exists. Currently we get the following error with the AllPluginSmokeTest

```
Traceback (most recent call last):
  File "/tmp/cirrus-ci-build/sos/report/__init__.py", line 1251, in setup
    plug.setup()
  File "/tmp/cirrus-ci-build/sos/report/plugins/charmed_postgresql.py", line 140, in setup
    f"topology {self.patroni_cluster_name}"),
                ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/cirrus-ci-build/sos/report/plugins/charmed_postgresql.py", line 41, in patroni_cluster_name
    with open(f"{PATHS['PATRONI_CONF']}/patroni.yaml") as file:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/var/snap/charmed-postgresql/current/etc/patroni/patroni.yaml'
```

Reply:
Setup is redesigned. Now, we collect the files always. If the config file exists and structured correctly, additional commands run for better insight. Also, password is directly set on the process' environment so that it does not leak on verbose flag. This is the same with the regular PostgreSQL plugin.